e2etest/l2tests/assignment.go

```diff
@@ -479,5 +479,67 @@ var _ = ginkgo.Describe("IP Assignment", func() {
 				&svc3.Status.LoadBalancer.Ingress[0]))
 			framework.ExpectNoError(err)
 		})
+
+		ginkgo.It("with namespace update", func() {
+			secondNamespacePool := metallbv1beta1.IPAddressPool{
+				ObjectMeta: metav1.ObjectMeta{Name: "second-ns-labels-ip-pool"},
+				Spec: metallbv1beta1.IPAddressPoolSpec{
+					Addresses: []string{
+						"192.168.30.0/30",
+					},
+					AllocateTo: &metallbv1beta1.ServiceAllocation{Priority: 10, NamespaceSelectors: []metav1.LabelSelector{{MatchLabels: secondNsLabels}}},
+				},
+			}
+
+			newLabels := make(map[string]string)
+			for i, j := range secondNsLabels {
+				newLabels[i] = j
+			}
+			newLabels["newLabel"] = "true"
+
+			secondNamespacePoolHigherPriority := metallbv1beta1.IPAddressPool{
+				ObjectMeta: metav1.ObjectMeta{Name: "second-ns-labels-higher-priority-ip-pool"},
+				Spec: metallbv1beta1.IPAddressPoolSpec{
+					Addresses: []string{
+						"192.168.40.0/32",
+					},
+					AllocateTo: &metallbv1beta1.ServiceAllocation{Priority: 5, NamespaceSelectors: []metav1.LabelSelector{{MatchLabels: newLabels}}},
+				},
+			}
+
+			resources := internalconfig.ClusterResources{
+				Pools: []metallbv1beta1.IPAddressPool{secondNamespacePool, secondNamespacePoolHigherPriority},
+			}
+			err := ConfigUpdater.Update(resources)
+			framework.ExpectNoError(err)
+
+			svc1, _ := service.CreateWithBackend(cs, secondNamespace, "second-ns-service")
+			defer func() {
+				service.Delete(cs, svc1)
+			}()
+
+			ginkgo.By("validate LoadBalancer IP is allocated from 1st address pool")
+			err = config.ValidateIPInRange([]metallbv1beta1.IPAddressPool{secondNamespacePool}, e2eservice.GetIngressPoint(
+				&svc1.Status.LoadBalancer.Ingress[0]))
+			framework.ExpectNoError(err)
+
+			ginkgo.By("updating second namespace labels to match higher priority pool")
```

Comment:
would it make sense to have this as a step of one of the existing tests? This would save a bit of test running time

Reply:
done